cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -287,22 +288,47 @@ def __init__(self, arg_list=None):
         # if user not provided, give a default user
         if not self.run_as_user and not self._arg_parser.upgrade:
             self.run_as_user = self.EXTERNAL_USER_NAME
-        if not self._arg_parser.rgw_pool_prefix and not self._arg_parser.upgrade:
-            self._arg_parser.rgw_pool_prefix = self.DEFAULT_RGW_POOL_PREFIX
         if self.ceph_conf:
             self.cluster = rados.Rados(conffile=self.ceph_conf)
         else:
             self.cluster = rados.Rados()
             self.cluster.conf_read_file()
         self.cluster.connect()
 
+        if not self._arg_parser.rgw_pool_prefix and not self._arg_parser.upgrade:
+            # auto detect prefix
+            self._arg_parser.rgw_pool_prefix = self.detect_rgw_pool_prefix()
+
     def shutdown(self):
         if self.cluster.state == "connected":
             self.cluster.shutdown()
 
     def get_fsid(self):
         return str(self.cluster.get_fsid())
 
+    def detect_rgw_pool_prefix(self):
+        cmd_json = {"prefix": "osd lspools", "format": "json"}
+        ret_val, json_out, err_msg = self._common_cmd_json_gen(cmd_json)
+        # if there is an unsuccessful attempt,
+        if ret_val != 0 or len(json_out) == 0:
+            raise ExecutionFailureException(
+                "'osd lspools' command failed.\n" +
+                "Error: {}".format(err_msg if ret_val != 0 else self.EMPTY_OUTPUT_LIST))
+
+        found = False
+        for pool in json_out:
+            if '.rgw.meta' in pool['poolname']:
+                pool = pool['poolname']  # type: str
+                # extract prefix name
+                rc = re.compile(r'.+?(?=\.rgw\.meta)')
+                self._arg_parser.rgw_pool_prefix = rc.findall(pool)[0]
+                found = True
+                return self._arg_parser.rgw_pool_prefix
```

Comment:
Do we require `found` bool variable as we are returning immediately after finding the rgw-prefix?